axum-macros/src/debug_handler.rs

```diff
@@ -60,6 +63,67 @@ fn check_extractor_count(item_fn: &ItemFn) -> syn::Result<()> {
     }
 }
 
+fn extractor_idents(item_fn: &ItemFn) -> impl Iterator<Item = (usize, &syn::FnArg, &syn::Ident)> {
+    item_fn
+        .sig
+        .inputs
+        .iter()
+        .enumerate()
+        .filter_map(|(idx, fn_arg)| match fn_arg {
+            FnArg::Receiver(_) => None,
+            FnArg::Typed(pat_type) => {
+                if let Type::Path(type_path) = &*pat_type.ty {
+                    type_path
+                        .path
+                        .segments
+                        .last()
+                        .map(|segment| (idx, fn_arg, &segment.ident))
+                } else {
+                    None
+                }
+            }
+        })
+}
+
+fn check_request_last_extractor(item_fn: &ItemFn) -> syn::Result<()> {
+    let request_extractor_ident =
+        extractor_idents(item_fn).find(|(_, _, ident)| *ident == "Request");
+
+    if let Some((idx, fn_arg, _)) = request_extractor_ident {
+        if idx != item_fn.sig.inputs.len() - 1 {
+            return Err(syn::Error::new_spanned(
+                fn_arg,
+                "`Request` extractor should always be last",
+            ));
+        }
+    }
+
+    Ok(())
+}
+
+fn check_path_extractor(item_fn: &ItemFn) -> TokenStream {
+    let path_extractors = extractor_idents(item_fn)
+        .filter(|(_, _, ident)| *ident == "Path")
+        .collect::<Vec<_>>();
+
+    if path_extractors.len() > 1 {
+        path_extractors
+            .into_iter()
+            .map(|(_, arg, _)| {
+                syn::Error::new_spanned(
+                    arg,
+                    "Multiple parameters must be extracted with a tuple \
+                    `Path<(_, _)>` or a struct `Path<YourParams>`, not by applying \
+                    multiple `Path<_>` extractors",
+                )
+                .to_compile_error()
+            })
+            .collect()
```

Comment:
Instead of emitting multiple `compile_error!`s here, it might be better to build up one `syn::Error` from the path extractors using `syn::Error::combine` (i.e. one time this error message, then "other path extractor here") and turn that into a `compile_error!`.

Reply:
Do you mean something like:

```rust
let mut errors = path_extractors
    .into_iter()
    .map(|(_, arg, _)| {
        syn::Error::new_spanned(
            arg,
            "Multiple parameters must be extracted with a tuple \
            `Path<(_, _)>` or a struct `Path<YourParams>`, not by applying \
            multiple `Path<_>` extractors",
        )
    })
    .collect::<Vec<_>>();
let mut error = errors.remove(0);
error.extend(errors); // extend calls `syn::Error::combine`
error.to_compile_error()
```

because that doesn't change the output.

Reply:
I guess the output might be different on nightly? Anyways, if it's always separate errors on stable even with `Error::combine`, then I guess it doesn't make a lot of sense.